src/components/facet-line/FacetLine.tsx

```diff
@@ -25,8 +31,10 @@ const FacetLine: React.FunctionComponent<FacetLineProps> = ({ q }) => {
 
   return (
     <>
-      {!data && <FacetLineFiltersSkeleton />}
-      {data && <FacetLineFilters facets={data.search.intelligentFacets} />}
+      {(isLoading || isLoadingFacets) && <FacetLineFiltersSkeleton />}
+      {data && hitCount > 0 && (
+        <FacetLineFilters facets={data.search.intelligentFacets} />
+      )}
```

Comment:
Please consider if passing data about the number of hits is really necessary for the facet line to render the facet line.

As I see it it passes information about the state of the search result (`hitcount` and `isLoading`) which should not really be necessary for rendering the facet line. I think the situation might be caused by the `useIntelligentFacetsQuery()` always returning the same result. That should no longer be the case now. Thus I would think that if the search has 0 results then there will be no facets to render in the line either causing it to be empty.

If that holds true then we should be able to reduce this to using a `isLoadingFacets` variable to determine whether we should render a skeleton or the result.

Instead of passing `hitcount` and `isLoading` down we should be able to use these to control whether subcomponents are rendered or not in the `SearchResult` component.

Reply:
You're completely right. I made it more difficult than it needed to be.